examples/99-advanced/reorder-edges.py

```diff
@@ -0,0 +1,97 @@
+"""
+.. _reorder_edges_example:
+
+Reorder Edges
+~~~~~~~~~~~~~
+
+This example demonstrates how to reorder edges based on connectivity.
+
+Boundary edges are extracted from a :class:`pyvista.PolyData` are unordered. We
+can use a simple mapping array based on :attr:`lines <pyvista.PolyData.lines>`
+to reorder the edges to make them sequential.
+
+"""
+
+import numpy as np
+
+import pyvista as pv
+from pyvista import examples
+
+###############################################################################
+# Clip dataset
+# ~~~~~~~~~~~~
+# Download an example dataset and clip it so we have open edges using the
+# :func:`clip <pyvista.DataSet.clip>` filter.
+dataset = examples.download_bunny_coarse()
+mesh = dataset.clip('y', invert=True)
+mesh.plot()
+
+
+###############################################################################
+# Extract and reorder edges
+# ~~~~~~~~~~~~~~~~~~~~~~~~~
+# Extract the edges and show they are unordered.
+#
+
+edge_data = mesh.extract_feature_edges(
+    boundary_edges=True, feature_edges=False, manifold_edges=False
+)
+
+# show the edges are unordered
+pl = pv.Plotter()
+pl.add_mesh(mesh)
+pl.add_mesh(
+    edge_data,
+    scalars=range(edge_data.n_points),
+    line_width=10,
+    show_scalar_bar=False,
+    cmap='plasma',
+)
+pl.show()
+
+
+###############################################################################
+# Reorder edges
+# ~~~~~~~~~~~~~
+# Reorder the edges and plot it.
+
+edges = edge_data.lines.reshape(-1, 3)[:, 1:]  # n_edges x 2
+u, v = edges.T
+
+# create a simple adjcency mapper array
+adj = np.empty(u.shape, dtype=u.dtype)
+adj[u] = v
+
+# Walk through each adjacent edge.
+v = edges[0, 0]  # Starting vertex
+vert_idxs = [v]
+for _ in range(len(edges)):
+    v = adj[v]
+    vert_idxs.append(v)
```

Comment:
I don't have much spare bandwidth at the moment, but this looks to me like it assumes that every line is oriented in the same way. In other words, every point along the closed (mangled) curve appears in the first position exactly once and in the second position exactly once.

In other words, it assumes we don't have lines like `[..., 2, 41, 42, 2, 43, 42]` in which case point `42` is never in the first position and the corresponding value in `adj` is left garbage.

The foolproof solution would be something like using [darikg's dict-based solution](https://github.com/pyvista/pyvista/discussions/3211#discussioncomment-3475803), defining adjacency in both directions (both `u -> v` and `v -> u`) and in the adjacency loop keeping track of seen points in a set.

It's plausible that surface orientation somehow leads to edges always being ordered in the same direction, but I don't know if we can take this for granted.

Reply:
Quite right. This algorithm breaks down if there isn't a single closed loop. Multiple loops don't work, only the first loop gets stored (and repeated because we iterate based on the number of edges).